students/HABTAMU/lesson02/assignment/code/clean_up.py

```diff
@@ -0,0 +1,24 @@
+"""
+this module clean the json source data,
+when end rental date is ealier than start rental date.
+
+    Args: take source.json as an input file.
+    Return: correct_source.json as a cleaned output json file.
+
+"""
+
+import json
+import datetime
+
+with open('source.json', 'r') as input:
+     data = json.load(input)
+     for value in data.values():
+             rental_start = datetime.datetime.strptime(value['rental_start'], '%m/%d/%y')
+             rental_end = datetime.datetime.strptime(value['rental_end'], '%m/%d/%y')
+             if (rental_end - rental_start).days < 0:
+                     temp_value = value['rental_start']
+                     value['rental_start'] = value['rental_end']
+                     value['rental_end'] = temp_value
+
+with open('correct_source.json', 'w', encoding='utf-8') as output:
+     output.write(json.dumps(data, indent=4,ensure_ascii=False))
```

Comment:
I like the approach of having a separate cleaning data module. You could have separate cleaning modules for different customers or data, and then you could assume that your data is clean in your other module, and you can verify that you are doing the right thing with your customer with everything you are doing in one place. Nice! I would make it into a function, however. Makes it easier to test, and you do want tests on it.